### DISCUSSION TITLE
Dogecoin Moving Away from PoW (researching an alternative implementation)

### DISCUSSION THREAD

**Discussion by magneticat** [dogecoin/dogecoin#2331] [General]
Hello everybody.

I apologize if this contains little technical information, as I know that core developers generally dislike this (rightfully so).

I would like to open a discussion about future improvements (post 1.21.x) in Dogecoin, which might also involve more drastic rewrites, that could address the _possibility_ of Dogecoin moving away from PoW and towards other solutions - hybrid or completely detached from mining.

While I believe in the value of PoW mining, as I wrote in a past issue that was closed a while ago (and that was concerning a more generic road map discussion which was considered useless), I believe that there will soon be a strong shift in public opinion and lawmakers against cryptos that involve massive use of electricity. I think that in the long term this might turn out to be a huge problem for Doge, and having this discussion now might be as critical as when we discussed about merged mining in 2014. While Bitcoin can (maybe) live on its fame alone and locked in a safe like the gold in Fort Knox, Doge needs to work as a currency, needs to be fast, cheap, and it needs to be associated with "doing good". A full frontal attack against PoW cryptos might be the end of Doge, not just because of eventual bans on mining, but because it will involve also a moral judgment on its perceived wastefulness (again, I am not saying this would be a completely fair assumption).

In this discussion, I would like to gather the opinions of current developers and people with technical knowledge and blockchain development experience on this topic.

To a person like me, with zero experience in blockchain development, a possible solution would seem to be forking another crypto. There are several that are very fast, feeless, but of course none of them had to face the same popularity and level of use that Doge has reached. Creating a fork of a crypto like nano for example (which has a lot in common with Doge, being designed to be a working currency; one of nano's promininent community members had created an issue about this: https://github.com/dogecoin/dogecoin/issues/2024), and allowing, via a bridge, to transfer liquidity from Doge into it, and then back in the other direction. So, at first, this would coexist with the current Dogecoin network and implementation, but we would have a "Plan B" ready in case the situation starts going south.

There are other cryptos out there that could offer a similar possibility, so I would be happy to hear from the people in the github community that are familiar with them.

Forking an existing crypto would obviously come also with the possibility of a backlash - saying that we are just rehashing somebody else's code. Maybe such an effort could be done in cooperation with developers of the other crypto to allow a bridge into this new blockchain also from the original crypto that is being forked.

We could setup a tipjar to help support this "research".

If this is inappropriate, please feel free to remove/close the thread. And developers and coders, feel free to demolish this idea. I just wanted to have a discussion about this, and I know I lack the knowledge for a more advanced discussion which this would require.

**Comment by hollabigj** [comment]
I don't have experience in developing blockchain. However, I think it's a great idea to move away from PoW to PoS that would drives the valuation of Dogecoin up. Although, that would require Doge to hire blockchain developers or someone willing to put their work in their free time.

If dogecoin only takes less than or equals to 1 kwh for mining, perhaps go hybrid instead? The gov would appreciates more with low energy consumption on mining.

It's just my two cents, not in my expertise opinion.

**Reply by CHIPofficial** [reply]
this might be hard nicoll said it may take a long time.

**Reply by hollabigj** [reply]
I remember he mentioned about that and I agreed with him. It takes a while to build a "perfect" blockchain.

It would be nice to have PoS on their roadmap eventually. Perhaps, get the investor(s) like Elon Musk, Mark Cuban or asking Dogecoin community approval to allocate 1% of the $ pool (not sure it's legal to do so) to invest in the blockchain developers that would speed up the process. It's difficult to develop the blockchain on their free time.

It's a smart business if they do smart contract like Ethereum and go hybrid or PoS.

However, I'm fine with what Dogecoin developers think what is the best for the Dogecoin. I am cool with that.

**Reply by jsrobson10** [reply]
1 kwh isn't too much right now compared to more mined cryptocurrencies like bitcoin and etherium, but as dogecoin gains in popularity there will be more miners. more miners mean a much more secure network by driving up the price of a 51% attack by increasing the difficulty per block. This will make mining less profitable but also more secure.

Dogecoin uses approx 120W per transaction ([source](https://www.gfinityesports.com/cryptocurrency/dogecoin-environmental-impact-energy-consumption-environmentally-friendly-cryptocurrency/)). If you multiply this by 18749 (number of transactions in last 24 hours), you get approx 2249 KW per day. I had an idea (in a different discussion) where dogecoin is still based on PoW, but the work is done by transactions instead of by miners to solve the issue of fighting over who has the next part of the chain and to remove the financial incentive.

**Comment by crevolutionX** [comment]
I've got an idea of a hybrid PoW/PoS system that could potentially solve the energy consumption problem. I'd propose that we implement a second layer of security so the network could be secured by both PoW and PoS. So this is the idea..

A new consensus rule would allow the network to reject a fork that currently has the longest PoW (so the hodlers could indirectly vote for the "true" ledger). There would be three little steps to sign up, vote then agree on the real blockchain. This new system works that way:

Step 1: all the users who want to participate must create a key pair then link their wallets' addresses to their public key (the owner of the wallets must authorize the operation with his signature). In the voting process the amount of votes one voter has is proportional to the amount of coins that he owns (the votes must include the voter's signature to be valid).

Step 2: Any participant can create a poll and allow the network to vote in favor of rejecting a specific block. Any user who doesn’t answer a poll is automatically voting in favor of not rejecting the block (even if he is not registered), the majority vote should then decide whether the block is accepted or rejected by the network (the last block of the fork should be used as a reference point for counting the votes).

Step 3: The participants must update their own record and share their data (it doesn't matter if all the records are not exactly the same as it will probably not affect the polls’ result). Each node operator must set his own voting parameters.

The incentive for a node operator to participate is to protect the wealth of his clients. The goal is to remove the block reward then secure the network with a PoS sidechain.

Let me know what you think. Thanks!

**Comment by AbcSxyZ** [comment]
To add to the discussion, found some @patricklodder comments about PoW switch on [dogecoindev subreddit](https://www.reddit.com/r/dogecoindev/comments/o94p6j/fee_policy_change_proposal/h3by9nq?utm_source=share&utm_medium=web2x&context=3).

> **yanki2del :**
> Guys, without individual wallets moving to proof of stake is impossible. What can we do when 70% of all Dogecoins is in Robinhood's single wallet?
>
> **patricklodder :**
> Ask RH "when", because they kinda said they'd do chain integration. Also, even when that happens, doing PoS will have many many issues and it's debatable whether it is needed.
>
> **magneticat :**
>   "Also, even when that happens, doing PoS will have many many issues and it's debatable whether it is needed."
> 
> Indeed. I support PoW and I believe PoW for emission and a layer 2 on top of it would cover our use case wonderfully, but unfortunately I am also seriously afraid of regulations and bans against crypto mining in the future, which could force some PoW cryptos into extinction or to market caps that are so low they become too prone to price manipulation.
>
> So it is debatable... but a debate to be had, maybe after 1.21 is released? I would LOVE people with high level of experience and knowledge about transitioning from PoW to PoS or other solutions to provide some feedback on this, also to help ME understand whether my concerns in researching an alternative are legitimate.
>
> Thanks for all your work Patrick.
>
> **patricklodder :**
> I'd personally be really glad to discuss later, because, after using 1.14 for the critical path to enable better fees, 1.21 should be the focus and it needs tons and tons of QA.
> 
> It's good for a discussion to happen, just it cannot be my personal priority. I really don't have the time to think about this right now because that kind of things require focus and it would make me drop other things that I think are more urgent. I have your discussion topic saved though - will get to it eventually ❤️

**Comment by jsrobson10** [comment]
I think moving away from PoW and moving to Proof of Stake would be great. It would be a good idea to avoid other ones like Proof of Space, or any other hardware based proofs, so that people don't create lots of "miners". I like the idea of proof of stake because it makes 1 miner with the same amount of funds at least no different to multiple with funds spread among all of them, so using more hardware is a waste. 

For DOGE to be more "Do Only Good Everyday", it should move to Proof of Stake to save our environment and drive the prices of hardware down. Maybe it could be a hybrid network at first, people will move to staking as they don't have to pay for power anymore, then maybe after a set date mining could be completely removed.

**Reply by patricklodder** [reply]
Grave concerns over asset classification aside, proof of stake (especially in modern delegated form) make sense when you have a savings asset, because you need to have a large portion of the coin illiquid and "at stake", to protect against a bad actor accumulating a significant amount of coin. This would make sense for a coin that one would want to hoard and change hands as little as possible. However, for a coin that aims to be a currency, it makes not much sense to make such a large part illiquid. So this may not be ideal.

**Reply by jsrobson10** [reply]
A bad actor could get alot of dogecoin and stake that if not much doge is being staked, but how is this much different/worse from a bad actor getting access to alot of processing power, in the case of a 51% attack? It's probably better in some ways because it means a bad actor can't do a 51% attack with a botnet.

**Reply by patricklodder** [reply]
Because of limited supply. A good actor can counter a bad actor with a botnet of his own, but if a bad actor owns the majority of all coins, there is no way to counter it.

**Reply by magneticat** [reply]
Hey @patricklodder , but isn't this how people used to use savings accounts? Same currency, but a part of it went into a savings account and a part in your spending account. That would be a lot better than do what people are doing now - staking Dogecoin in centralized exchanges that (as written also in their terms of services) will use your holdings to lend them to large investors, which can then misuse those funds to manipulate the price.

I do not see staking as something against a spending currency, TBH. The big problem I see (in addition to asset classification which you mentioned) is that some realities we have now, such as large exchange wallets, would have enormous power in a PoS system. And I have no idea how you would move from a PoW to a PoS (or hybrid) system.

PoW's decentralization and simplicity is a great thing, of course, so I know how problematic this discussion can be.

Thanks a lot for all the replies and the discussion. I have been in a lot of troubles in the past months and missed them.

**Reply by patricklodder** [reply]
>  I have been in a lot of troubles in the past months and missed them.

Sorry to hear that! Hope you're ok now. If not, DM me on reddit 🙂 

Quick reply to your other points:

> staking Dogecoin in centralized exchanges

That's a misnomer. It's not staking, it's borrowing.

> I do not see staking as something against a spending currency, TBH. 

It really is a problem because liquidity becomes the enemy of security that way. See https://arxiv.org/abs/2006.11156 for example.

**Reply by magneticat** [reply]
> That's a misnomer. It's not staking, it's borrowing.

Of course! Thanks for pointing that out.

Thanks for that link as well!

**Reply by jsrobson10** [reply]
thanks for this discussion! you've changed my mind about proof of stake. based on this discussion I don't think moving to PoS would be a good idea.

**Comment by jsrobson10** [comment]
I wonder if it would be a good idea for Dogecoin to loosely follow the direction that IOTA is going. Maybe dogecoin could migrate to a hybrid blockchain/transaction chain where transactions on the chain validate other transactions. If this were to happen, then fees wouldn't be going to miners, so maybe there might be no fee, or fees could just be destroyed to prevent spam. 

This would require some pretty big modifications to the dogecoin core client, but if it's for a more sustainable coin then it might be a good idea. This would mean dogecoins mining process wouldn't be where coins are minted. Maybe the supply could become finite and locked at some fixed value, or maybe coins could be minted through some other process, like operating a full node or through a system where hodlers gain interest. 

Maybe there could still be a blockchain but the minute interval could be handled by the next transaction and regulated by number of transactions in a block instead of block mining difficulty. To prevent nodes fighting over who has the longest chain, a bit of CPU hashing could be done by the next transaction to work on the blockhash and that could be the deciding factor of who has the next part of the chain.  

So this would still be based on proof of work, but the work is done by transactions instead of by miners, so there's no financial incentive (maybe even a financial disincentive if fees are destroyed) to fill a warehouse full of NVIDIA RTX 3090s to generate doge.

**Comment by Zerophase** [comment]
Isn't PoS less secure than PoW? The coin is also centralized in a few wallets mostly.

**Reply by patricklodder** [reply]
Correct, but shibes are just looking out for CZ's and Vlad's retirement :joy:

Seriously though, I currently do not see any path forward on PoS specifically, but I've agreed with OP that once we have more time on our hands, I'll spend some time going over proposed alternatives, because I don't see anyone truly looking into assessing them.

**Reply by magneticat** [reply]
Thanks @patricklodder!  And I know this is not high priority right now!

**Comment by andreww14111** [comment]
GREAT
